libs/counterfactuals/src/lib/CounterfactualComponentUtils.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+enum FieldChangeUpdate {
+  Dither = "dither",
+  Property = "property",
+  Type = "type"
+}
+
+export function isJustTypeChange(changedKeys: string[]): boolean {
+  // return true only if type of the axis has changed in panel
+  const changedKeysTemp = removeParentKeys(changedKeys);
+  if (
+    changedKeysTemp.length === 1 &&
+    changedKeysTemp.includes(FieldChangeUpdate.Type)
+  ) {
+    return true;
+  }
+  return false;
+}
+
+function removeParentKeys(changedKeys: string[]): string[] {
+  const valuesToRemove = new Set(["options", "xAxis", "yAxis"]); // Since chartProps is a nested object, these are parent keys which are usually changed if inner keys are changed.
+  return changedKeys.filter((item) => !valuesToRemove.has(item));
+}
```

Comment:
What does "parent" refer to here?

Reply:
It is a nested obeject. For ex:
const a = {name: "xyz",
address: {
street: "123 st",
area: "wer"},
dob:"123223"}

here address is parent key, as if area has changed, then address has also changed.